lib/java/com/google/android/material/drawable/DrawableUtils.java

```diff
@@ -205,44 +228,51 @@ public static Drawable compositeTwoLayeredDrawable(
     }
     LayerDrawable drawable =
         new LayerDrawable(new Drawable[] {bottomLayerDrawable, topLayerDrawable});
+
+    final int topLayerWidth = topLayerDrawable.getIntrinsicWidth();
+    final int topLayerHeight = topLayerDrawable.getIntrinsicHeight();
+    final int bottomLayerWidth = bottomLayerDrawable.getIntrinsicWidth();
+    final int bottomLayerHeight = bottomLayerDrawable.getIntrinsicHeight();
+
     int topLayerNewWidth;
     int topLayerNewHeight;
-    if (topLayerDrawable.getIntrinsicWidth() == -1 || topLayerDrawable.getIntrinsicHeight() == -1) {
-      // If there's no intrinsic width or height, keep bottom layer's size.
-      topLayerNewWidth = bottomLayerDrawable.getIntrinsicWidth();
-      topLayerNewHeight = bottomLayerDrawable.getIntrinsicHeight();
-    } else if (topLayerDrawable.getIntrinsicWidth() <= bottomLayerDrawable.getIntrinsicWidth()
-        && topLayerDrawable.getIntrinsicHeight() <= bottomLayerDrawable.getIntrinsicHeight()) {
-      // If the top layer is smaller than the bottom layer in both its width and height, keep top
-      // layer's size.
-      topLayerNewWidth = topLayerDrawable.getIntrinsicWidth();
-      topLayerNewHeight = topLayerDrawable.getIntrinsicHeight();
+    if (topLayerDesiredWidth != -1 && topLayerDesiredHeight != -1) {
```

Comment:
Is there currently a case where topLayerDesiredWidth will be -1? Looks like the default set in MaterialSwitch's constructor is 0, which is technically a valid size. What case is this -1 check trying to handle?

Reply:
`compositeTwoLayeredDrawable` is also used in `MaterialCheckBox`.

Reply:
👍